File: python/S1_BASIC/L4_BASICMATHS/checkpalindrome.py
class Solution:
    def __init__(self,num):
        self.num=num
    def reverse(self):
        r=0
        temp=self.num
        while temp>0:
            r=r*10+(temp%10)
            temp//=10
        return r
    def isPalindrome(self):
        r=self.reverse()
        return r==self.num
    
if __name__=="__main__":
    num=int(input("enter number :"))
    s=Solution(num)
    print(s.reverse())
    print(s.isPalindrome())

File: JAVA/S1_BASIC/L4_BASICMATHS/checkpalindrome.java
class solution{
    int  reverse(int n){
        int num=n;
        int r=0;
        while(num>0){
            r=r*10+num%10;
            num/=10;
        }
        return r;
    }
    int isPalindrome(int num){
        int n=reverse(num);
        if(n==num)return 1;
        return 0;
    }
}
public class checkpalindrome{
    public static void main(String[] args){
    solution s=new solution();
    System.out.println(s.isPalindrome(121));
    }

}